src/domain/LottoGameController.js

```diff
@@ -1,119 +1,115 @@
 
 export default class LottoGameController {
   #lottoGame;
 
   constructor() {
     this.#lottoGame = new LottoGame();
+    this.purchaseView = new PurchaseView();
+    this.lottoListView = new LottoListView();
+    this.winningNumbersView = new WinningNumbersView();
+    this.modalView = new ModalView();
   }
 
-  async play() {
-    await this.#createLotto();
-    await this.#compareLotto();
-    await this.#processRestart();
+  listenViewEvents() {
+    this.purchaseView.$purchaseAmountForm.addEventListener('purchase', (e) =>
+      this.#getPurchaseAmount(e.detail)
+    );
+    this.winningNumbersView.$winningLottoForm.addEventListener('check', (e) =>
+      this.#getWinningLottos(e.detail.winningNumbers, e.detail.bonusNumber)
+    );
+    this.modalView.$retryButton.addEventListener('retry', () => {
+      this.#processRestart();
+    });
   }
 
-  async #createLotto() {
-    const purchaseAmount = await this.#requestPurchaseAmount();
-    const lottoQuantity = purchaseAmount / LOTTO_CONDITION.lottoPrice;
+  #getPurchaseAmount(purchaseAmount) {
+    this.#lottoGame.resetLottoGame();
+    this.#createLotto(purchaseAmount);
+  }
 
-    const eachLottoNumbers = Array.from({ length: lottoQuantity }, () => {
-      const lottoNumbers = this.#lottoGame.generateLottoNumbers(LOTTO_CONDITION.lottoDigits);
-      this.#lottoGame.makeLotto(lottoNumbers);
+  #createLotto(purchaseAmount) {
+    try {
+      Validation.validatePurchaseAmount(purchaseAmount);
 
-      return lottoNumbers;
-    });
+      const lottoQuantity = purchaseAmount / LOTTO_CONDITION.lottoPrice;
+      const eachLottoNumbers = Array.from({ length: lottoQuantity }, () => {
+        const lottoNumbers = this.#lottoGame.generateLottoNumbers(LOTTO_CONDITION.lottoDigits);
 
-    OutputView.printLottoQuantity(lottoQuantity);
-    OutputView.printEachLottoNumbers(eachLottoNumbers);
-  }
+        this.#lottoGame.makeLotto(lottoNumbers);
 
-  async #compareLotto() {
-    const winningNumbers = await this.#requestWinningNumbers();
-    const bonusNumber = await this.#requestBonusNumber(winningNumbers);
+        return lottoNumbers;
+      });
 
-    const eachCompareResult = this.#lottoGame.getEachCompareResult(winningNumbers, bonusNumber);
-    const statistics = this.#lottoGame.getStatistics(eachCompareResult);
-    const totalPrizeMoney = this.#lottoGame.getTotalPrizeMoney(statistics);
-    const yieldRatio = this.#lottoGame.getYieldRatio(totalPrizeMoney);
+      this.#onValidPurchaseAmount(lottoQuantity, eachLottoNumbers);
 
-    OutputView.printStatistics(statistics);
-    OutputView.printYieldRatio(yieldRatio);
-  }
+      return;
+    } catch (error) {
+      if (error instanceof Error) {
+        this.#onInValidPurchaseAmount(error.message);
 
-  async #processRestart() {
-    const command = await this.#requestRestartCommand();
+        return;
+      }
 
-    if (command === RESTART_COMMAND.quit) {
-      Console.close();
-      return;
+      return window.alert(error);
```

Comment:
```suggestion
     window.alert(error);
     return;
```

window.alert는 아무것도 반환하지 않거니와 그 값을 사용하지도 않기 때문에 
따로 써주는게 더 좋을거 같아요